test/test_common/utility.cc

```diff
@@ -311,16 +311,22 @@ std::string TestUtility::convertTime(const std::string& input, const std::string
 }
 
 // static
-bool TestUtility::gaugesZeroed(const std::vector<Stats::GaugeSharedPtr>& gauges) {
-  // Returns true if all gauges are 0 except the circuit_breaker remaining resource
+std::string TestUtility::nonZeroedGauges(const std::vector<Stats::GaugeSharedPtr>& gauges) {
```

Comment:
seems like the only use of this function can be better expressed with the old function?

Reply:
This version of the function prints out which stat was non-zero and what it's value was when the test expectation fails (this function doesn't actually print anything, but the test expectation itself is `EXPECT_EQ(empty_string, nonZeroedGauges())`, which prints the string when it isn't empty). With the old version, I had to attach a debugger to get that information, so I found this version much easier to iterate on test failures with.

Reply:
There may be a better gtest/gmock way to do this, and I played with returning a vector of something that the expectation could print out, but it was more work than it was worth (to me) to figure out how to make it print nicely (not print pointers; print names of stats), so I just did this slightly clunky version with a string.